### PR TITLE
IW-4970 | Deploy to artifactory

### PR DESCRIPTION
**Pull request by kasperekt**
I've prepared some changes for design-system. Thanks to this PR we are able to use artifcatory for this package. The package itself is available with `@fandom/design-system` name.

We also switched from building SVG sprite and SCSS inside ember CLI tools to separate scripts (thanks to `sass` and `svgstore` packages).

## Next Steps
* Prepare Jenkins pipeline(s) for auto deployment after merge to master and for building documentation (building and pushing it to separate branch)
* Remove `docs` from master and keep it inside `gh-pages` branch

### CODE REVIEW

**Review by latata — APPROVED** [review]

File: Jenkinsfile-auto-deploy [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+def publishMajorLabel = "publish-major-on-merge"
+def deployDocsLabel = "deploy-docs-on-merge"
+
+node('qa-executors') {
+	stage('Clone repo and validate') {
+		nodejs('v10 LTS') {
+			git credentialsId: 'bd3cf712-39a7-4b16-979e-ff86208ab2ea', url: 'git@github.com:Wikia/design-system.git', branch: "master"
+			currentBuild.description = "commit: ${params.commit}, PR: ${params.pull_num}"
+
+			stage('Check Pull Request labels') {
```
Comment (nikodamn) [inline Jenkinsfile-auto-deploy:19]: If the job was triggered by the webhook `pull_request` event type it should receive the labels in the webhook data. See for example: [webhook](https://github.com/Wikia/pandora/blob/master/service/github-webhooks-proxy/src/main/resources/github-webhooks-proxy.yaml#L429) and [corresponding job](http://jenkins.wikia-prod:8080/job/feeds-and-posts-auto-deploy-marker/6843/parameters/). If it is the case for you here you may want to add the `labels` job parameterer and there'd be no need to curl it.

Although if it works and you are fine with it, we can keep it. ;)

File: Jenkinsfile-auto-deploy [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+				labels = parsed.name
+				println "Labels found: " + labels
+			}
+
+			withEnv(["ARTIFACTORY_NPM_TOKEN=${env.JENKINS_ARTIFACTORY_PASSWORD}"]) {
+				stage('Install dependencies') {
+					try {
+						sh 'yarn install'
+					} catch (e) {
+						error 'FAIL'
```
Comment (nikodamn) [inline Jenkinsfile-auto-deploy:31]: Can we add some more descriptive error message there?

File: Jenkinsfile-auto-deploy [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+					}
+				}
+
+				stage('Run tests') {
+					try {
+						sh 'yarn test:jenkins'
+					} catch (e) {
+						error 'FAIL'
```
Comment (nikodamn) [inline Jenkinsfile-auto-deploy:39]: Can we add some more descriptive error message there?

File: Jenkinsfile-auto-deploy [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+					}
+				}
+
+				if (currentBuild.currentResult.equals("SUCCESS")) {
+					if (labels.contains(publishMajorLabel)) {
```
Comment (nikodamn) [inline Jenkinsfile-auto-deploy:44]: In general this is good, although I'm thinking what should happend when someone adds two or three labels. It's kind of an edge case, but might be worth considering what should happen.

File: Jenkinsfile-auto-deploy-marker [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+node('linux') {
+	try {
+		stage('Add label to GitHub repo') {
+			def githubUsername = "WikiaJenkins"
+
+			def publishMinorLabel = "publish-minor-on-merge"
+			def deployDocsLabel = "deploy-docs-on-merge"
+
+			if (!labels.contains(publishMinorLabel)) {
+				sh "curl -X POST ${authHeader} ${contentTypeHeader} --data '[ \"${publishMinorLabel}\" ]' ${issueUrl}/labels"
```
Comment (nikodamn) [inline Jenkinsfile-auto-deploy-marker:15]: Is there a need for this curl here? Labels are already provided in the `labels` parameter and they should come with the webhook event

File: Jenkinsfile-auto-deploy (hunk 4 above)
Comment (kasperekt) [reply Jenkinsfile-auto-deploy:44]: It's an edge-case but it's handled but this set of if/elseif blocks. The priorities are:
1. major
2. minor
3. patch
So if we have all 3 labels then it'll release major version.

File: Jenkinsfile-auto-deploy-marker [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+				def message = "**Watch out!** This PR will start deployment to **artifactory**.\\n \\n If you do not want it, please remove the `publish-minor-on-merge` label to disable it. Keep in mind that this **label will be re-applied if new commits will be added to this PR**."
+				sh "curl -X POST ${authHeader} ${contentTypeHeader} --data '{ \"body\": \"${message}\"}' ${issueUrl}/comments"
+			}
+
+			if (!labels.contains(deployDocsLabel)) {
+				sh "curl -X POST ${authHeader} ${contentTypeHeader} --data '[ \"${deployDocsLabel}\" ]' ${issueUrl}/labels"
```
Comment (nikodamn) [inline Jenkinsfile-auto-deploy-marker:22]: Same about this one and `labels` parameter

File: Jenkinsfile-auto-deploy-marker [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+			def githubUsername = "WikiaJenkins"
+
+			def publishMinorLabel = "publish-minor-on-merge"
+			def deployDocsLabel = "deploy-docs-on-merge"
+
+			if (!labels.contains(publishMinorLabel)) {
+				sh "curl -X POST ${authHeader} ${contentTypeHeader} --data '[ \"${publishMinorLabel}\" ]' ${issueUrl}/labels"
+
+				def message = "**Watch out!** This PR will start deployment to **artifactory**.\\n \\n If you do not want it, please remove the `publish-minor-on-merge` label to disable it. Keep in mind that this **label will be re-applied if new commits will be added to this PR**."
+				sh "curl -X POST ${authHeader} ${contentTypeHeader} --data '{ \"body\": \"${message}\"}' ${issueUrl}/comments"
```
Comment (nikodamn) [inline Jenkinsfile-auto-deploy-marker:18]: Developing leftover?

**Review by nikodamn — APPROVED** [review]
Looks good to me! Left few minor suggestions

File: Jenkinsfile-auto-deploy-marker (hunk 5 above)
Comment (kasperekt) [reply Jenkinsfile-auto-deploy-marker:15]: This is a POST request, so it'll change the labels.

File: Jenkinsfile-auto-deploy-marker (hunk 6 above)
Comment (kasperekt) [reply Jenkinsfile-auto-deploy-marker:22]: Like above. It's a POST request

File: Jenkinsfile-auto-deploy-marker (hunk 7 above)
Comment (kasperekt) [reply Jenkinsfile-auto-deploy-marker:18]: Not really, why is that? It's a POST request for creating a comment that'll warn the creator that the PR will be merged (or docs will be deployed) on merge to master.

Comment (nikodamn) [reply Jenkinsfile-auto-deploy-marker:18]: Yeah, nevermind, I thought it's GET :)

**Review by nikodamn — APPROVED** [review]

**Pull request merged**